libshpool/src/config.rs

```diff
@@ -212,6 +237,37 @@ pub struct Config {
     pub motd_args: Option<Vec<String>>,
 }
 
+impl Config {
+    // Merge with `another` Config instance, with `self` taking higher priority.
+    //
+    // Top level options with simple value are directly taken from the higher priority instance.
+    // The merge strategy for list or map are handled case by case. Please refer to each options'
+    // documentation for details.
+    pub fn or(self, another: Config) -> Config {
+        Config {
+            norc: self.norc.or(another.norc),
+            noecho: self.noecho.or(another.noecho),
+            nosymlink_ssh_auth_sock: self
+                .nosymlink_ssh_auth_sock
+                .or(another.nosymlink_ssh_auth_sock),
+            noread_etc_environment: self.noread_etc_environment.or(another.noread_etc_environment),
+            shell: self.shell.or(another.shell),
+            // TODO: Check this
+            env: self.env.or(another.env),
+            forward_env: self.forward_env.or(another.forward_env),
+            initial_path: self.initial_path.or(another.initial_path),
+            session_restore_mode: self.session_restore_mode.or(another.session_restore_mode),
+            output_spool_lines: self.output_spool_lines.or(another.output_spool_lines),
+            // TODO: check this
```

Comment:
dangling?

Reply:
Resolve the other two. Will update once we settle down on a merge strategy for these fields.